### ISSUE TITLE
QUESTION: Shouldn't OTBR_BACKBONE_ROUTER=ON should also configure OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE=1?

### ISSUE THREAD

**Issue by AlanLCollins** [openthread/ot-br-posix#1966]
OTBR_BACKBONE_ROUTER=ON is currently modifying OTBR_ENABLE_BACKBONE_ROUTER=1, but OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE remains default value = 0. 

so even that otbr-agent is adding BBR support, OT does not support BBR which includes important features like Multicast forwarding to be OFF.

**Comment by abtink** [comment]
My understanding is that `OTBR` CMake options (as defined in  https://github.com/openthread/ot-br-posix/blob/main/etc/cmake/options.cmake) are to configure OTBR itself. 

OT core has its own set of configs. Of course there may be certain dependancies between the two.

**Comment by gabekassel** [comment]
@abtink can you comment on how this is typically managed (e.g. such as in your products or the OTBR demo)? Do you have the BBR define in a custom config? It feels misleading to enable BBR from OTBR cmake flag but it's not actually on in OT core.

**Comment by abtink** [comment]
I'm not sure how our team handles this today.

In general, the idea is that a product build can integrate a bunch of components and libraries (ot-core, otbr, platform, drivers, crypto lib, app/next layer code, etc), each with its own set of configurations.

In earlier products, we used to use configuration headers to customize everything. I personally like/prefer this model
- We had a separate configuration header for each component, e.g., `openthread-core-<product-name>-config.h`. We passed this header to the OT build using `OT_PROJ_CONFIG` or `OPENTHREAD_PROJECT_CORE_CONFIG_FILE`. This header defined all of the OT core configurations needed for that specific product (or set of products).
- In some cases, we had a product-level configuration header with its own set of high-level product-specific features to enable or disable. We then used this header to determine the configurations for each component (e.g. if high-level "Jam Detection" features was enabled on product, we enabled the related OT config, and app/UI configs for it).
- The nice thing about this model is that if we need to figure out what OT configurations were used for a specific build or product, we can easily look them up in one header file. We don't have to search through different makefiles.
